pkg/webhook/pod/mutating/pod_create_update_handler.go

```diff
@@ -46,31 +46,49 @@ var _ admission.Handler = &PodCreateHandler{}
 // Handle handles admission requests.
 func (h *PodCreateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
 	obj := &corev1.Pod{}
-
-	err := h.Decoder.Decode(req, obj)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	var copy runtime.Object = obj.DeepCopy()
-	// when pod.namespace is empty, using req.namespace
-	if obj.Namespace == "" {
-		obj.Namespace = req.Namespace
-	}
-
-	injectPodReadinessGate(req, obj)
-
-	err = h.sidecarsetMutatingPod(ctx, req, obj)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-	if reflect.DeepEqual(obj, copy) {
+	var err error
+
+	switch req.AdmissionRequest.Operation {
+	case admissionv1beta1.Create, admissionv1beta1.Update:
```

Comment:
Don't put pod update/delete hook here. (Actually the resource for update operation you want to watch is `pods/status`, rather than `pods`)

Pod mutating:
- pods create

Pod validating:
- pods delete
- pods/eviction create
- pods/status update

Reply:
The thing I also worry about is that hook all pods/status update might have bad effect on the user's cluster.

Reply:
Indeed, we just care aboute the Pods with a specfic annotation in `pods/status `update, maybe the influence is minor.